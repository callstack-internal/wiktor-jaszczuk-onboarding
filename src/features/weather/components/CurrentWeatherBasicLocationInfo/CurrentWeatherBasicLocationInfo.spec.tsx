import {render} from '@testing-library/react-native';
import type {Weather} from '../../hooks/useGetWeatherForLocation';
import type {OWMetrics} from '../../services/openWeatherApiService/types';
import {CurrentWeatherBasicLocationInfo} from './index';

const METRICS: OWMetrics[] = ['metric', 'imperial', 'standard'];

describe('CurrentWeatherBasicLocationInfo', () => {
  it.each(METRICS)('Given UnitType: %s, then it will match snapshot', unit => {
    expect(
      render(
        <CurrentWeatherBasicLocationInfo weather={WEATHER_MOCK} unit={unit} />,
      ),
    ).toMatchSnapshot();
  });
});

const WEATHER_MOCK: Weather = {
  clouds: {
    all: 0,
  },
  coord: {
    lat: 51.11,
    lon: 17.03,
  },
  dt: 1741185257,
  main: {
    feels_like: 15.48,
    grnd_level: 1008,
    humidity: 23,
    pressure: 1024,
    sea_level: 1024,
    temp: 17.11,
    temp_max: 17.21,
    temp_min: 16.69,
  },
  sys: {
    country: 'PL',
    sunrise: 1741152487,
    sunset: 1741192736,
  },
  timezone: 3600,
  visibility: 10000,
  weather: [
    {
      description: 'clear sky',
      icon: '01d',
      id: 800,
      main: 'Clear',
    },
  ],
  wind: {
    deg: 270,
    speed: 2.57,
  },
  local_names: {
    ar: 'فروتزواف',
    be: 'Уроцлаў',
    bg: 'Вроцлав',
    cs: 'Vratislav',
    de: 'Breslau',
    en: 'Wrocław',
    eo: 'Vroclavo',
    es: 'Breslavia',
    fr: 'Vratislavie',
    he: 'ורוצלב',
    hu: 'Boroszló',
    it: 'Breslavia',
    ja: 'ヴロツワフ',
    la: 'Wratislavia',
    lt: 'Vroclavas',
    lv: 'Vroclava',
    mk: 'Вроцлав',
    pl: 'Wrocław',
    pt: 'Breslávia',
    ru: 'Вроцлав',
    sk: 'Vroclav',
    sr: 'Вроцлав',
    uk: 'Вроцлав',
    yi: 'ברעסלוי',
    zh: '弗罗茨瓦夫',
  },
  name: 'Wrocław',
  country: 'PL',
  state: 'Lower Silesian Voivodeship',
};
