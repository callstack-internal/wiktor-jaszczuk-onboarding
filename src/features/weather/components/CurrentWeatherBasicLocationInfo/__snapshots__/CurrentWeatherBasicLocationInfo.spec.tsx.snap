// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeatherBasicLocationInfo Given UnitType: imperial, then it will match snapshot 1`] = `
<View
  style={
    [
      undefined,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
      },
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://openweathermap.org/img/wn/01d@2x.png",
      }
    }
    style={
      {
        "aspectRatio": 1,
        "height": "100%",
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
        }
      }
    >
      Wrocław
    </Text>
    <Text>
      clear sky
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignSelf": "center",
        },
        {
          "alignItems": "center",
          "aspectRatio": 2.5,
          "backgroundColor": "cyan",
          "justifyContent": "center",
        },
        {
          "borderRadius": 20,
          "height": 40,
          "padding": 10,
        },
      ]
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        {
          "fontSize": 20,
          "lineHeight": 20,
        }
      }
    >
      17.1
      °F
    </Text>
  </View>
</View>
`;

exports[`CurrentWeatherBasicLocationInfo Given UnitType: metric, then it will match snapshot 1`] = `
<View
  style={
    [
      undefined,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
      },
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://openweathermap.org/img/wn/01d@2x.png",
      }
    }
    style={
      {
        "aspectRatio": 1,
        "height": "100%",
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
        }
      }
    >
      Loading...
    </Text>
    <Text>
      clear sky
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignSelf": "center",
        },
        {
          "alignItems": "center",
          "aspectRatio": 2.5,
          "backgroundColor": "cyan",
          "justifyContent": "center",
        },
        {
          "borderRadius": 20,
          "height": 40,
          "padding": 10,
        },
      ]
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        {
          "fontSize": 20,
          "lineHeight": 20,
        }
      }
    >
      17.1
      °C
    </Text>
  </View>
</View>
`;

exports[`CurrentWeatherBasicLocationInfo Given UnitType: standard, then it will match snapshot 1`] = `
<View
  style={
    [
      undefined,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
      },
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://openweathermap.org/img/wn/01d@2x.png",
      }
    }
    style={
      {
        "aspectRatio": 1,
        "height": "100%",
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
        }
      }
    >
      Wrocław
    </Text>
    <Text>
      clear sky
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignSelf": "center",
        },
        {
          "alignItems": "center",
          "aspectRatio": 2.5,
          "backgroundColor": "cyan",
          "justifyContent": "center",
        },
        {
          "borderRadius": 20,
          "height": 40,
          "padding": 10,
        },
      ]
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        {
          "fontSize": 20,
          "lineHeight": 20,
        }
      }
    >
      17.1
      K
    </Text>
  </View>
</View>
`;
